ravenframework/Optimizers/Optimizer.py

```diff
@@ -131,7 +132,8 @@ def getInputSpecification(cls):
         descr=r"""name of \xmlNode{Function} which contains implicit constraints of the Model. From a practical
               point of view, this XML node must contain the name of a function defined in the \xmlNode{Functions}
               block (see Section~\ref{sec:functions}). This external function must contain a method called
-              ``implicitConstrain'', which returns True for outputs satisfying the implicit constraints and False otherwise.""")
+              ``implicitConstraint'', which returns a float value which should
+        be less than zero when the constraint is violated.""")
```

Comment:
Josh, this will be applied to all Optimizers. Unfortunately the "float" value  is valid only for the GA, the gradient descent uses a rejection boolean. So this would make now the "Gradient Descent" explanation wrong.

Reply:
You are right, changed to only affect GA.